community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/index/LuceneIndexAccessor.java

```diff
@@ -172,35 +174,26 @@ private LuceneIndexUpdater( boolean inRecovery )
         }
 
         @Override
-        public void process( NodePropertyUpdate update ) throws IOException
+        public PreparedIndexUpdates prepare( Iterable<NodePropertyUpdate> updates ) throws IOException
         {
-            switch ( update.getUpdateMode() )
+            int additions = 0;
+            for ( NodePropertyUpdate update : updates )
             {
-                case ADDED:
-                    if ( inRecovery )
-                    {
-                        addRecovered( update.getNodeId(), update.getValueAfter() );
-                    }
-                    else
-                    {
-                        add( update.getNodeId(), update.getValueAfter() );
-                    }
-                    break;
-                case CHANGED:
-                    change( update.getNodeId(), update.getValueAfter() );
-                    break;
-                case REMOVED:
-                    LuceneIndexAccessor.this.remove( update.getNodeId() );
-                    break;
-                default:
-                    throw new UnsupportedOperationException();
+                // Only count additions, since removals will not affect the size of the index until it is merged
```

Comment:
Turns out it would as it is an atomic remove+add. Will include `CHANGED` into reservation count.